src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj

```diff
@@ -94,22 +94,28 @@
   uses. JARS_NO_REQUIRE was the legacy way to turn off jar loading but is
   being phased out in favor of JARS_REQUIRE.  As of JRuby 1.7.20, only
   JARS_NO_REQUIRE is honored.  Setting both of those here for forward
-  compatibility."
+  compatibility.
+
+  We also merge an environment-vars map with the config to allow for whitelisted
+  environment variables to be visible to the Ruby code. This map is by default
+  set to {} if the user does not specify it in the configuration file."
   [env :- jruby-schemas/EnvMap
-   gem-home :- schema/Str]
+   gem-home :- schema/Str
+   config :- (schema/pred map?)]
   (let [whitelist ["HOME" "PATH"]
         clean-env (select-keys env whitelist)]
-    (assoc clean-env
-      "GEM_HOME" gem-home
-      "JARS_NO_REQUIRE" "true"
-      "JARS_REQUIRE" "false")))
+    (merge (assoc clean-env
```

Comment:
@cprice404 I can't remember if we talked about our preferences for this before or not but do you think users should be able to use the custom environment variable configuration capability to override the special ones like `HOME`, `PATH`, and `GEM_HOME` that we define by default?  I think I'm okay with saying that they can do that, like @janelu2's implementation would do currently, so long as we document the ones that we have special values for so that users understand that they might cause problems if they choose to override our built-in defaults.  What do you think?